spec.md

```diff
@@ -51,12 +51,14 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
 interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
 
 For clarity, when a feature is marked as "OPTIONAL" this means that it is
-OPTIONAL for both the sender and receiver of a message to support that feature.
-In other words, a sender can choose to include that feature in a message if it
-wants, and a receiver can choose to support that feature if it wants. A receiver
-that does not support that feature will then silently ignore that part of the
-message. The sender needs to be prepared for the situation where a receiver
-ignores that feature.
+OPTIONAL for both the [Producer](#producer) and [Consumer](#consumer) of a
+message to support that feature. In other words, a producer can choose to
+include that feature in a message if it wants, and a consumer can choose to
+support that feature if it wants. A consumer that does not support that feature
+will then silently ignore that part of the message. The producer needs to be
+prepared for the situation where a consumer ignores that feature. For an
```

Comment:
I know this isn't part of your PR, but I'm struggling with what this sentence is trying to say.

The producer should be loosely coupled from the consumer anyway. That makes the producer always prepared for the consumer ignoring parts of the message (whether OPTIONAL or not) or the whole message, doesn't it? 😉

Reply:
As you already said, I didn't write this. 😃 
Could it mean that without the OPTIONAL attribute the consumer would interpret the event differently?

Reply:
Normally that kind of language is important when the specs have a processing model associated with it. In those cases a REQUIRED property typically can not be ignored by the receiver if they want to be compliant with the spec because there's some processing rules associated with it. While an OPTIONAL property can be ignored, which means the sender needs to be prepared for that.

In our case, we don't have anything as strong as a processing model that people need to adhere to, but I do think people need to be prepared for cases such as when they add "partionkey" to events and assume that means the receiver will put all incoming events into the same partition, but in fact that may not happen. The receiver can completely ignore it.